### PR TITLE
fix #116196: Crash applying line via double click (debug build)

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

**Comment by AntonioBL** [comment]
Well, this PR does not completely solve the problem. If you try to add a hairpin with a double click from the palette you still hit the Q_ASSERT at line 737 of libmscore\element.h (referring to line 791 of element.h); the ottava case is solved since with this PR it no more goes inside the if-clause of line 294 of libmscore\cmd.cpp and therefore does not hit the "TextLine* tl = toTextLine(spanner)" instruction.
Hairpins, pedals and voltae are still affected, but only when inserted with double click from the palette.

**Comment by Jojo-Schmitz** [comment]
Change to HairpinSegment and VoltaSegment? There os no PedalSegment though

**Comment by Jojo-Schmitz** [comment]
Change to HairpinSegment and VoltaSegment seems to fix that. There is no PedalSegment though, not sure what to do about this?

**Comment by Jojo-Schmitz** [comment]
Ah, found PedalSegment and PEDAL_SEGMENT

**Comment by AntonioBL** [comment]
But that means it is now just completely skipping the  if-clause of line 294 of cmd.cpp. Should it be removed then?
Otherwise, if the spanner is not strictly a textLine, the "toTextLine" instruction will _always_ give the Q_ASSERT.

**Comment by Jojo-Schmitz** [comment]
Hmm, just reverting back to `TextLine* tl = static_cast<TextLine*>(spanner);` should fix this too then?

**Comment by Jojo-Schmitz** [comment]
OK, let's try another Approach then

File: libmscore/element.h [hunk 1]
```diff
@@ -732,6 +734,18 @@ static inline const DurationElement* toDurationElement(const Element* e) {
          || e->type() == Element::Type::REPEAT_MEASURE || e->type() == Element::Type::TUPLET);
       return (const DurationElement*)e;
       }
+static inline TextLine* toTextLine(Element* e) {
```
Comment (Jojo-Schmitz) [inline libmscore/element.h]: maybe `toSortOfTextLine()` instead?

**Comment by Jojo-Schmitz** [comment]
Seems this got made obsolete by 0b92b20 ?

**Comment by lasconic** [comment]
After a weekend of thinking, Werner changed his mind and he believe "that in this special case a dynamic_cast may be faster and its the intended semantic". See https://github.com/musescore/MuseScore/commit/0b92b2055ee3e612f73510ea4c67a26bf5d807b1
Would you mind doing another PR for the toNoteLine part ?

**Comment by Jojo-Schmitz** [comment]
Done, see #2735, but it isn't used anywhere

**Pull request closed without merging**